File: src/main/Process.java
package main;

public class Process {

    public int pid;
    public int arrivalTime;
    public int burstTime;
    public int priority;

    public int waitingTime;
    public int turnaroundTime;
    public int remainingTime;
    public int startTime;
    public int completionTime;

    public Process(int pid, int arrivalTime, int burstTime, int priority) {
        this.pid = pid;
        this.arrivalTime = arrivalTime;
        this.burstTime = burstTime;
        this.priority = priority;
        this.remainingTime = burstTime;
        this.startTime = -1;
        this.completionTime = 0;
    }

    public Process copy() {
        return new Process(pid, arrivalTime, burstTime, priority);
    }
}

File: gui/src/types/process.js
/**
 * Process type for CPU scheduling simulation
 */
export const createProcess = (pid, arrivalTime, burstTime, priority) => ({
  pid,
  arrivalTime,
  burstTime,
  priority,
  waitingTime: 0,
  turnaroundTime: 0,
  remainingTime: burstTime,
  startTime: -1,
  completionTime: 0,
})

export const copyProcess = (p) => createProcess(p.pid, p.arrivalTime, p.burstTime, p.priority)
